ACE/ace/Functor.h

```diff
@@ -200,6 +200,32 @@ class ACE_Export ACE_Hash<unsigned long>
   unsigned long operator () (unsigned long t) const;
 };
 
+#if (ACE_SIZEOF_LONG == 8)
+/**
+ * @brief Function object for hashing a long long number
+ */
+template<>
+class ACE_Export ACE_Hash<long long>
```

Comment:
What about providing a specialization for all ACE_(U)INTxx defines, looks we do that for ACE_Equal in this file already

Reply:
See #1091 for a separate PR to try that